hazelcast/src/main/java/com/hazelcast/cache/impl/AbstractCacheService.java

```diff
@@ -288,6 +289,7 @@ public void deleteCache(String name, String callerUuid, boolean destroy) {
         CacheConfig config = deleteCacheConfig(name);
         if (destroy) {
             destroySegments(name);
+            sendInvalidationEvent(name, null, SOURCE_NOT_AVAILABLE);
```

Comment:
deleteCache is also being called from https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/cache/impl/operation/CacheDestroyOperation.java#L61 with flag set to true and I guess it will have duplicate events in this case.

I just debugged the test case when_cacheDestroyed_invalidationEventIsReceived test case and I see that the CacheDestroyMessageTask and CacheDestroyOperation is not being called. This looks like another bug. Instead of this, the implementation calls DestroyProxyMessageTask at the server side, this does not look correct.

I think due to this bug, you are seeing a single invalidation, if this operation actually worked it would be multiple such events.